features/environment.py

```diff
@@ -81,20 +85,15 @@ def reduce_db_outcomes(feature_outcomes):
             if failed_outcomes:
                 unique_outcomes = set() # TODO __hash__ + __eq__ will be better
                 unique_objects = [obj for obj in failed_outcomes if get_validation_outcome_hash(obj) not in unique_outcomes and (unique_outcomes.add(get_validation_outcome_hash(obj)) or True)]
-                return unique_objects
-
+                yield from unique_objects
             else:
                 outcome_counts = Counter(outcome.severity for outcome in context.gherkin_outcomes)
-
                 for severity in [OutcomeSeverity.PASSED, OutcomeSeverity.EXECUTED, OutcomeSeverity.NOT_APPLICABLE]:
                     if outcome_counts[severity] > 0:
-                        for outcome in context.gherkin_outcomes:
-                            if outcome.severity == severity:
-                                return [outcome]
-
-            return []
+                        yield next(o for o in context.gherkin_outcomes if o.severity == severity)
```

Comment:
```suggestion
                        yield next(outcome for outcome in context.gherkin_outcomes if outcome.severity == severity)
```
For clarity and to match with what we previously had